pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
 		return err
 	}
-	deploymentList := "\n"
-	for _, deployment := range deployments.Items {
-		deploymentList += fmt.Sprintf("%s\n", deployment.Name)
+	ctx.Printlnf("Starting to delete the pods")
+	//delete pods
+	for _, pod := range pods.Items {
+		pod := pod // TODO We won't need it after upgrading to go 1.22: https://go.dev/blog/loopvar-preview
+		if err := cl.Delete(ctx, &pod); err != nil {
+			return err
+		}
 	}
-	term.PrintContextSeparatorWithBodyf(deploymentList, "Existing deployments in %s namespace", ns)
 	return nil
+
 }
 
-func scaleToZero(cl runtimeclient.Client, namespacedName types.NamespacedName) (int32, error) {
-	// get the deployment
-	deployment := &appsv1.Deployment{}
-	if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-		return 0, err
+func restartNonOperatorDeployments(ctx *clicontext.CommandContext, deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+
+	o := kubectlrollout.NewRolloutRestartOptions(ioStreams)
+
+	if err := o.Complete(f, nil, []string{"deployment"}); err != nil {
+		panic(err)
 	}
-	// keep original number of replicas so we can bring it back
-	originalReplicas := *deployment.Spec.Replicas
-	zero := int32(0)
-	deployment.Spec.Replicas = &zero
 
-	// update the deployment so it scales to zero
-	return originalReplicas, cl.Update(context.TODO(), deployment)
+	o.Resources = []string{"deployment/" + deployment.Name}
+
+	if err := o.Validate(); err != nil {
+		panic(err)
+	}
+	ctx.Printlnf("Running the rollout restart command for non-Operator deployment %v", deployment.Name)
+	return o.RunRestart()
 }
 
-func scaleBack(term ioutils.Terminal, cl runtimeclient.Client, namespacedName types.NamespacedName, originalReplicas int32) error {
-	return wait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
-		term.Println("")
-		term.Printlnf("Trying to scale the deployment back to '%d'", originalReplicas)
-		// get the updated
-		deployment := &appsv1.Deployment{}
-		if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-			return false, err
-		}
-		// check if the replicas number wasn't already reset by a controller
-		if *deployment.Spec.Replicas == originalReplicas {
-			return true, nil
-		}
-		// set the original
-		deployment.Spec.Replicas = &originalReplicas
-		// and update to scale back
-		if err := cl.Update(context.TODO(), deployment); err != nil {
-			term.Printlnf("error updating Deployment '%s': %s. Will retry again...", namespacedName.Name, err.Error())
-			return false, nil
-		}
-		return true, nil
-	})
+func checkRolloutStatus(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error {
+
+	cmd := kubectlrollout.NewRolloutStatusOptions(ioStreams)
+
+	if err := cmd.Complete(f, []string{"deployment"}); err != nil {
+		panic(err)
+	}
+	cmd.LabelSelector = labelSelector
+	if err := cmd.Validate(); err != nil {
+		panic(err)
```

Comment:
why `panic` and not returning the error?

Reply:
if we return an error, the function would stop there.. if there was an error in completing/validating the command, it not running the command, we do not want to stop just for those error,  and we do want to go ahead checking the status, we still want to go ahead and do rollout restart for next deployment  .. that were my thoughts

Reply:
I don't follow, panic will stop the execution immediately, right? 
TBH, if something like this fails, then we should return an error - there is apparently something wrong (maybe some bug), but we should do it properly by returning an error, not by killing it.

Reply:
Okay i updated it now , 
previously i meant that it will still allow the parent function to go on with panic. but your suggestion makes sense